lms/static/sass/multicourse/_dashboard.scss

```diff
@@ -347,357 +280,387 @@
+        position: relative;
+        top: 50%;
+        transform: translateY(-50%);
 
-            span {
-              font-weight: bold;
-            }
-          }
-        }
+        // UI: course item actions
 
-        .enter-course {
-          @include button(simple, $button-color);
+        .action {
           @include box-sizing(border-box);
+          @include margin-right($baseline/2);
+          display: inline-block;
+          float: right;
+          min-width: ($baseline*2);
+          color: $gray-l3;
           border-radius: 3px;
-          display: block;
-          @include float(left);
-          font: normal 15px/1.6rem $sans-serif;
-          letter-spacing: 0;
-          padding: 6px 32px 7px;
-          text-align: center;
-          margin-top: 16px;
+          padding: 12px;
+          border: 1px solid $white;
+          
+          &:hover, &:focus {
+            color: #333;
+            border: 1px solid $gray-l4;
+          }
 
-          &.archived {
-            @include button(simple, $button-archive-color);
-            font: normal 15px/1.6rem $sans-serif;
-            padding: 6px 32px 7px;
+          // STATE: is-disabled
+          &.is-disabled {
+            color: #808080;
+            color: $gray-l4;
+          }
 
-            &:hover, &:focus {
-              text-decoration: none;
-            }
+          // TYPE: facebook share
+          &.action-facebook, {
+            color: #3b5998;
           }
 
-          &:hover, &:focus {
-            text-decoration: none;
+          // TYPE: twitter share
+          &.action-twitter {
+            color: #55ACEE;
           }
         }
-      }
 
-      .prerequisites {
-        @include clearfix;
+        // UI: general actions dropdown layout
+        .wrapper-action-more {
+          display: inline-block;
+          position: relative;
+          float: right;
 
-        .tip {
-          font-family: $sans-serif;
-          font-size: 1em;
-          color: $lighter-base-font-color;
-          margin-top: ($baseline/2);
-        }
-      }
 
-      // "enrolled as" status
-      .sts-enrollment {
-        position: absolute;
-        top: 105px;
-        @include left(0);
-        display: inline-block;
-        text-align: center;
-        width: 200px;
+          .actions-dropdown {
+            @extend %ui-no-list;
+            @extend %ui-depth1;
+            display: none;
+            position: absolute;
+            top: ($baseline*2);
+            right: 19px;
+            pointer-events: none;
+            min-width: ($baseline*8);
 
-        .label {
-          @extend %text-sr;
-        }
+            &.is-visible {
+              display: block;
+              pointer-events: auto;
+            }
 
-        .deco-graphic {
-          position: absolute;
-          top: -5px;
-          right: -8px;
-        }
+            .actions-dropdown-list {
+              @extend %ui-no-list;
+              display: table;
```

Comment:
I reused the same approach currently employed in the top right dropdown menu, which is part of the reason for the absolute positioning and other things within this menu. I can try and clean this up, though presently display: block doesn't render the dropdown wrapper as expected.